wakebox/main.cpp

```diff
@@ -73,12 +83,41 @@ int main(int argc, char *argv[])
 	}
 	ifs.close();
 
+	// Parse the params file as JSON.
+	JAST jast;
+	if (!JAST::parse(json, std::cerr, jast))
+		return 1;
+
+	// Extract the 'command' from the input json.
+	std::vector<char *> command;
+	for (auto &x : jast.get("command").children)
+		command.push_back(const_cast<char*>(x.second.value.c_str()));
+
+	// Extract the 'environment' from the input json.
+	std::vector<char *> env;
+	for (auto &x : jast.get("environment").children)
+		env.push_back(const_cast<char*>(x.second.value.c_str()));
+
+	// Handle --force-shell command line arg.
+	if (use_shell) {
+		use_stdin_file = false;
+		env.push_back(escaped_command_env(jast));
+
+		command.clear();
+		char sh[] = "/bin/sh";
+		command.push_back(sh);
+	}
+
+	// run_in_fuse() will pass the vector contents to exec as a null-terminated array.
+	command.push_back(0);
+	env.push_back(0);
```

Comment:
These should be moved to where we convert std::vector<std::string> to std::vector<const char*> right before the execve call.